lib/Sema/SemaBounds.cpp

```diff
@@ -524,25 +608,25 @@ namespace {
           MemberExpr *M = dyn_cast<MemberExpr>(E);
           if (!M) {
             llvm_unreachable("unexpected cast failure");
-            return CreateBoundsNone();
+            return CreateBoundsInferenceError();
           }
 
           FieldDecl *F = dyn_cast<FieldDecl>(M->getMemberDecl());
           if (!F)
-            return CreateBoundsNone();
+            return CreateBoundsInferenceError();
 
           BoundsExpr *B = F->getBoundsExpr();
           if (!B || B->isNone() || B->isInteropTypeAnnotation())
-            return CreateBoundsNone();
+            return CreateBoundsInferenceError();
 
           Expr *MemberBaseExpr = M->getBase();
           if (!SemaRef.CheckIsNonModifyingExpr(MemberBaseExpr,
               Sema::NonModifiyingExprRequirement::NMER_Unknown,
               /*ReportError=*/false))
-            return CreateBoundsNone();
+            return CreateBoundsAllowedButUncomputable();
```

Comment:
Again, another place where we now return `bounds(any)` instead of `bounds(none)` because we want to allow these expressions eventually, but can't express them in checked c today.